### PR TITLE
Add VertxContextSupport#executeBlocking()

### PR DESCRIPTION
**Pull request by mkouba**
- to improve WS next UX; i.e to run blocking code in HttpUpgradeCheck#perform() easily

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java [hunk 1]
```diff
@@ -96,14 +97,44 @@ public void accept(MultiSubscribe<T> ms) {
...
+     * If necessary, the CDI request context is activated during execution of the blocking code.
+     *
+     * @param <T>
+     * @param callable
+     * @return the produced {@link Uni}
+     * @see VertxContext#getOrCreateDuplicatedContext(Vertx)
+     */
+    public static <T> Uni<T> executeBlocking(Callable<T> callable) {
+        Context context = getContext(true);
+        return Uni.createFrom().completionStage(() -> {
```
Comment (cescoffier) [inline extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: defer might be slightly better.

@jponge wdyt?

**Review by cescoffier — APPROVED** [review]

File: extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java (hunk 1 above)
Comment (mkouba) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: > defer might be slightly better.

In what exactly? I mean, both suppliers are called at subscription time. And we would need to convert the `io.vertx.core.Future` to the `Uni` anyway.

Comment (cescoffier) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: I was referring to https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html which avoid using completion stages and it's slightly more optimized. But, yes, then you need an emitter, so the cost would be almost equivalent.

Comment (mkouba) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: > I was referring to https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html which avoid using completion stages and it's slightly more optimized.

Did you mean https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html#deferred(java.util.function.Supplier)?

Comment (cescoffier) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: Yes, sorry, missed the anchor.

Comment (jponge) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: Deferred with supplier is better, but a `CompletionStage` isn't incorrect per-se.

Comment (mkouba) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: > Deferred with supplier is better, but a `CompletionStage` isn't incorrect per-se.

Better in terms of performance, or?

Hm, but how do I create a `Uni` from `io.vertx.core.Future` without `Uni.createFrom().completionStage()`? That's what `io.smallrye.mutiny.vertx.UniHelper.toUni(Future<T>)` does by the way.

Comment (jponge) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java:112]: Yes yes you are right

**Pull request merged**